Combining Robert's code with mine

GJezierski opened the pull request:
*add login, logout
*add change, reset password on mail (in terminal Django we see this email send to reset password - reset link)
*add registration new users

GJezierski committed:
Combining Robert's code with mine
*add login, logout
*add change, reset password on mail (in terminal Django we see this email send to reset password - reset link)
*add registration new users

adam-harmasz commented on WeatherPortal/WeatherPortal/settings.py:
```diff
@@ -31,17 +31,18 @@
 # Application definition
 
 INSTALLED_APPS = [
+    # weatherapp must be first !!! must go before the admin application in INSTALLED_APPS
```
czemu?

GJezierski replied:
jak weatherapp był na dole INSTALLED_APPS, to opcja zmiany hasła prznosiły nas na adminowa zmiane hasła użytkownika, jak jest na górze, do przenosi nas na właściwą strone, jako że python przetwarza dane od góry do dołu, to wie żeby brać najpierw informacje z weatherapp a potem z  'django.contrib.admin'. Ostatecznie takie rozwiązanie tego problemu znalazlem na stackoverflow i działa :)

adam-harmasz replied:
Spojrzę dokładnie później, ale zawsze dawałem apki na dole i było ok

GJezierski replied:
https://stackoverflow.com/questions/447512/how-do-i-override-djangos-administrative-change-password-page

"If you still see the Django's Admin template, you must change the order of your INSTALLED_APPS (e.g if your template is inside some app, this app must appear before django.contrib.admin in INSTALLED_APPS)"

adam-harmasz replied:
No ok, ale pytanie brzmi czemu chcesz nadpisać zmianę hasła po stronie admina? To jest zmiana hasła tylko dla jakichś adminów, pracowników? Bo jeśli nie to powinien być jakichś dodatkowy endpoint do zmiany hasła i tam byłaby używana templatka jakaś Wasza, jak każdy inny endpoint, jest też już coś wbudowanego w Django

https://ccbv.co.uk/projects/Django/4.0/django.contrib.auth.views/PasswordChangeView/
https://ccbv.co.uk/projects/Django/4.0/django.contrib.auth.views/PasswordChangeDoneView/

Są też klasy do resetu hasła, można zobaczyć implementację bo Wy pewnie będziecie robić to podobnie, a może i warto byłoby skorzystać z tego?

adam-harmasz commented on WeatherPortal/WeatherPortal/settings.py:
```diff
@@ -85,26 +85,25 @@
-    {
-        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
-    },
+    # off safe passwords for test
+    # {
+    #     'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
+    # },
+    # {
+    #     'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
+    # },
+    # {
+    #     'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
+    # },
+    # {
+    #     'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
+    # },
```
Czemu nie chcecie walidatorów?

GJezierski replied:
Na koniec projektu włączymy walidatory, teraz poprastu dla prostoty wymyslania nowych hasel dla nowych uzytkownikow zostaly wylaczone

adam-harmasz replied:
To nie brzmi dobrze, te hasła nie muszą być mega trudne, wiem bo pracowałem nie raz przy włączonych walidatorach, a zakomentowany kod w pull requeście razi w oczy

adam-harmasz replied:
No i ogólnie wiadomo, że nie ma co sobie życia utrudniać, ale nie kosztem jakości

GJezierski replied:
Dzięki za radę to włącze walidatory i jeszcze raz pushne kod :)

adam-harmasz reviewed (commented):
O co chodzi w tym PR? Nie rozumiem tego `combining`, PRy można dodawać jeden po drugim i wszystko się dołącza do maina

GJezierski committed:
Combining Robert's code with mine
*add login, logout
*add change, reset password on mail (in terminal Django we see this email send to reset password - reset link)
*add registration new users

*on validation password
*add bootstrap buttons

robert-adamczyk commented:
U mnie działa wszystko
